### ISSUE TITLE
After training your model, the Rendered Image has no texture on it!

### ISSUE THREAD

**Issue by HOMGH** [tranluan/Nonlinear_Face_3DMM#56]
Hi @tranluan 
I could train your model by running **model_non_linear_3DMM.py** file. The rendered output has the correct shape and projection matrix, but it doesn't have any texture on it.  The rendered output is same as follows:

![rendered_image_05_0012](https://user-images.githubusercontent.com/65192528/88813070-51120f80-d16d-11ea-8dcf-7685146ca68c.jpg)
![rendered_image_01_0011](https://user-images.githubusercontent.com/65192528/88813113-60915880-d16d-11ea-80a8-e3ec1c7cfbd4.jpg)

I checked the warp_texture function and it is working, but it seems its input texture is not correct! That's why the output doesn't have any texture! The function used for texture generation is : **generator_decoder_albedo(tex_fx_300W, is_reuse=False, is_training=True)**. Is it the right function you used as your albedo decoder?

**Comment by chaoshiedwin** [comment]
I believe it is because 3DMM_tri_2d_barycoord.dat file is not correct, check out this issue: https://github.com/tranluan/Nonlinear_Face_3DMM/issues/27

**Comment by cassiePython** [comment]
Have you successfully corrected this issue? @chaoshiedwin @HOMGH

**Comment by HOMGH** [comment]
> Have you successfully corrected this issue? @chaoshiedwin @HOMGH

Yes, I just run the ....proxy.py file and it works. But still does not converge! (but at least results in something!)

**Comment by chaoshiedwin** [comment]
> Have you successfully corrected this issue? @chaoshiedwin @HOMGH

use 3DMM_tri_2d_barycoord_reduce.dat instead of 3DMM_tri_2d_barycoord.dat works for me.

**Comment by HOMGH** [comment]
> > Have you successfully corrected this issue? @chaoshiedwin @HOMGH
> 
> use 3DMM_tri_2d_barycoord_reduce.dat instead of 3DMM_tri_2d_barycoord.dat works for me.

Hi @chaoshiedwin , I actually gave up with that code and tried to run the **model_non_linear_3DMM_proxy.py**. It is on 11th epoch on 124k images but I think the results are not good . (attached images). Could you train the model?
![image](https://user-images.githubusercontent.com/65192528/92445122-babc0b00-f168-11ea-8684-fb3ec9311822.png)
![image](https://user-images.githubusercontent.com/65192528/92445180-cc9dae00-f168-11ea-8894-d632649bfd3b.png)

**Comment by huyanfei-cqupt** [comment]
> use 3DMM_tri_2d_barycoord_reduce.dat instead of 3DMM_tri_2d_barycoord.dat works for me.

Hi，@chaoshiedwin Did you get through this code?  Has the L0 loss converged in the pre-training phase?

**Comment by huyanfei-cqupt** [comment]
> > Have you successfully corrected this issue? @chaoshiedwin @HOMGH
> 
> use 3DMM_tri_2d_barycoord_reduce.dat instead of 3DMM_tri_2d_barycoord.dat works for me.

Hi,@chaoshiedwin, can you run the model_non_linear_3DMM.py successfully?  the G_loss_frecon how to write it？ any advices would be highly appreciated.
